src/modules/form/util/formUtil.ts

```diff
@@ -567,6 +567,17 @@ export const autofillValues = ({
     // Stop iterating through autofill values
     return true;
   });
+
+  // For readonly items, if no autofill condition matched, clear out the item's value.
+  // Don't make any changes if the whole form is in viewOnly mode, though, because
+  // in that case this is being called on form load (not on change of an autofill condition),
+  // so we don't want to clear any form values.
```

Comment:
Thanks, I think this is right. I tried to word it in another way: `For read-only items that are displayed on editable forms, the autofill value should nullify when it's conditions are no longer met. For example, a read-only field showing the sum of 2 input fields should be cleared if the inputs are cleared. (In that example, we assume the autofill rule for summing has an autofill_when condition requiring the inputs to be present)`. 
We're saying the same thing, right? Might be nice to our future selves to include the example in the code comment.

Reply:
Yes we are saying the same thing, and I think your comment is much clearer so I replaced mine 😂 thank you!